tile/platform/local_machine/program.h

```diff
@@ -8,6 +8,8 @@
 #include <set>
 #include <string>
 #include <unordered_map>
+#include <mutex>
```

Comment:
Please sort includes within sections.  (I'm surprised `pre-commit` isn't doing this automatically...?)